src/GitHub.Api/Git/RepositoryManager.cs

```diff
@@ -457,11 +425,16 @@ private void UpdateCurrentBranchAndRemote(string head)
                 }
             }
 
-            Logger.Trace("OnCurrentBranchUpdated: {0}", branch.HasValue ? branch.Value.ToString() : "[NULL]");
-            OnCurrentBranchUpdated?.Invoke(branch);
+            new ActionTask(taskManager.Token, () => {
+                Logger.Trace("OnCurrentBranchUpdated: {0}", branch.HasValue ? branch.Value.ToString() : "[NULL]");
+                OnCurrentBranchUpdated?.Invoke(branch);
 
-            Logger.Trace("OnCurrentRemoteUpdated: {0}", remote.HasValue ? remote.Value.ToString() : "[NULL]");
-            OnCurrentRemoteUpdated?.Invoke(remote);
+                Logger.Trace("OnCurrentRemoteUpdated: {0}", remote.HasValue ? remote.Value.ToString() : "[NULL]");
+                OnCurrentRemoteUpdated?.Invoke(remote);
+            })
+            {
+                Affinity = TaskAffinity.UI
+            }.Start();
```

Comment:
No other methods here switch to the UI thread for updating data, so this one probably shouldn't either. Maybe let Repository handle threading, have that logic all in one place?

Reply:
👍